### ISSUE TITLE
Morrighan detected by NGS.

### ISSUE THREAD

**Issue by Chaotix13567** [exectails/Morrighan#2]
When launching via MorriOneClick, Morrighan.exe is detected after several minutes by NGS. This was tested with Morrighan as the only modification to the game, and with other modifications that do not have issues (The current versions of Tiara and Abyss). Both resulted in detection by NGS when Morrighan was running.

**Comment by exectails** [comment]
Thanks for the report. I just logged into NA to test this though, and after 20 minutes I'm still able to play without any problems, with Morrighan as the only "mod", so I don't think that's the problem. Keep in mind that NGS is flaky at times, detecting things that aren't actually there. I've had situations where it kicked me a few times in a row, even without mods, before letting me play just fine *with* mods.

Has anyone else experienced this problem?

**Comment by Chaotix13567** [comment]
Interesting, I keep being detected, after a total of ~10 attempts. I am running the Steam version of Mabi and using the MorriOneClick that comes with mabicommerce alongside version 1.5.9 of Morrighan. Not sure if any of those are different from your testing; Perhaps it is something else on my side that I do not realize.

This is my log file for Morrighan over the past few attempts:

4/18/2017 3:37:24 AM [Info] - Morrighan 1.0.5862.36118
4/18/2017 3:37:24 AM [Info] - Starting server...
4/18/2017 3:37:24 AM [Info] - Server listening on '127.0.0.1:64857'
4/18/2017 3:37:24 AM [Info] - Starting Alissa communicator...
4/18/2017 3:37:24 AM [Info] - Starting client...
4/18/2017 3:37:26 AM [Info] - Started client: Client.exe code:1622 ver:249 chatip:208.85.109.37 chatport:8002 setting:"file://data/features.xml=Regular, USA" logip:127.0.0.1 logport:64857
4/18/2017 3:37:26 AM [Info] - Morrighan is ready to serve you
4/18/2017 3:37:57 AM [Info] - Client connected from 127.0.0.1:64861
4/18/2017 3:37:57 AM [Info] - Tunneling connection to '208.85.109.35:11000' through '127.0.0.1:64857'
4/18/2017 3:37:58 AM [Info] - Client closed connection
4/18/2017 3:37:58 AM [Info] - Server listening on '127.0.0.1:64857'
4/18/2017 3:38:07 AM [Info] - Client connected from 127.0.0.1:64864
4/18/2017 3:38:07 AM [Info] - Tunneling connection to '208.85.109.35:11000' through '127.0.0.1:64857'
4/18/2017 3:38:10 AM [Info] - Client closed connection
4/18/2017 3:38:10 AM [Info] - Server listening on '127.0.0.1:64857'
4/18/2017 3:38:10 AM [Info] - Client connected from 127.0.0.1:64866
4/18/2017 3:38:10 AM [Info] - Tunneling connection to '208.85.109.43:11020' through '127.0.0.1:64857'
4/18/2017 3:39:53 AM [Info] - Server closed connection
4/18/2017 3:39:53 AM [Info] - Server listening on '127.0.0.1:64857'

**Comment by exectails** [comment]
OneClick shouldn't make a difference, it simply starts Morrighan with the parameters for NA. I'm not using the Steam version, but that *shouldn't* make a difference either. I don't think Steam has anything to do with NGS.

**Comment by Chaotix13567** [comment]
https://github.com/Xcelled/mabicommerce/issues/16 This is a post I made on the mabicommerce thread. Since there are apparently no issues with either program, I am not sure why I am getting detected, or unable to run it correctly through PowerPatcher (Which is broken for updating I had thought, but not launching). I'm not really sure what else to try. Renaming MorriOneClick didn't change the results (I have been NGS'd before based on a filename), and I am unsure of how to change Morrighan.exe's name and have it still run.

**Comment by exectails** [comment]
Unfortunately I can't tell you what's being detected on your machine either, but I'm 99% sure that it's not Morrighan, seeing how nobody else mentioned such a problem to me and because I was able to play just fine.

Does NGS complain if you don't use Morrighan, starting it cleanly? And do you have any applications open that make use of Morrighan, like MabiCommerce or MabiPale?

**Comment by Chaotix13567** [comment]
I did a test where I started up a clean game, and after ~15 minutes of it running smoothly, I moved Morrighan.exe and MorriOneClick.exe into the game files, which shortly afterwards there was an NGS detection. I have been testing both with MabiCommerce/MabiPale and without. Both result in an NGS detection. I suspect that the file is being detected by name, which I believe is known with Cheat Engine and DATA Packer (I had a consistent NGS until I renamed it to DAT Packer). I got NGS'd with Morrighan in the files, but without running it, but not with a purely clean steam version.
